main.tf

```diff
@@ -135,18 +135,20 @@ module "gke" {
     max_memory_gb : var.cluster_autoscaling_max_memory_gb,
     min_cpu_cores : 1,
     min_memory_gb : 1,
-    gpu_resources = [],
-    auto_repair   = (var.kubernetes_channel == "UNSPECIFIED") ? false : true,
-    auto_upgrade  = (var.kubernetes_channel == "UNSPECIFIED") ? false : true
+    gpu_resources       = [],
+    auto_repair         = (var.kubernetes_channel == "UNSPECIFIED") ? false : true,
+    auto_upgrade        = (var.kubernetes_channel == "UNSPECIFIED") ? false : true
+    autoscaling_profile = var.cluster_autoscaling_profile
     } : {
     enabled : false,
     max_cpu_cores : 0,
     max_memory_gb : 0,
     min_cpu_cores : 0,
     min_memory_gb : 0,
-    gpu_resources = [],
-    auto_repair   = (var.kubernetes_channel == "UNSPECIFIED") ? false : true,
-    auto_upgrade  = (var.kubernetes_channel == "UNSPECIFIED") ? false : true
+    gpu_resources       = [],
+    auto_repair         = (var.kubernetes_channel == "UNSPECIFIED") ? false : true,
+    auto_upgrade        = (var.kubernetes_channel == "UNSPECIFIED") ? false : true
+    autoscaling_profile = var.cluster_autoscaling_profile
```

Comment:
As of `terraform-google-modules/kubernetes-engine/google//modules/private-cluster` v30.0.0, `autoscaling_profile` needs to now be defined in this map.